client/go/build.sh

```diff
@@ -0,0 +1,7 @@
+# Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+# Execute from this directory to build the command-line client to bin/vespa
+export GOPATH=`pwd`
+cd "$GOPATH/src/github.com/vespa-engine/vespa"
```

Comment:
I would be happy to avoid it, but are you sure, and how exactly? This is what https://golang.org/doc/gopath_code says to do, and if I don't set a $GOPATH I don't get a binary, and if I set $GOPATH it must be to the "workspace", which cannot be the directory having these files directly (it will then complain that there can't be a module file in a workspace.

Reply:
Ah, now I get it, keep the nested src but remove the other subdirs. Ok, thanks! They should update their doc ...

Reply:
It's a long story, but basically `GOPATH` is no longer necessary when modules are used. The source code can be located anywhere.

If `GOPATH` is unset it defaults to `~/go`, so if you run `go install` with `GOPATH` unset the binaries are created in `~/go/bin`. If you only want to control where binaries are created, you can set `GOBIN`.

Maybe easier explained with an example:
```
/tmp/example$ echo $GOPATH

/tmp/example$ grep -H '.*' *
go.mod:module github.com/mpolden/example
go.mod:
go.mod:go 1.16
main.go:package main
main.go:
main.go:import "fmt"
main.go:
main.go:func main() {
main.go:	fmt.Println("hello world")
main.go:}
/tmp/example$ go install
/tmp/example$ ls -l ~/go/bin/example
.rwxr-xr-x 2.0M martin 19 Aug 22:32 /Users/martin/go/bin/example
/tmp/example$ mkdir bin
/tmp/example$ export GOBIN=$PWD/bin
/tmp/example$ go install
/tmp/example$ ls -l bin/example
.rwxr-xr-x 2.0M martin 19 Aug 22:32 bin/example
```

Reply:
Yes, but how can I run the tests conveniently?
I worked around this now in build.sh but it's pretty lame.